### PR TITLE
Add Hilt dependency for Database auth module

### PR DESCRIPTION
**Pull request by ahominal**
- Added database authentication injection for classes & all test
- Refactored some tests (Transaction test need to be uncommented)!!
- Added Hilt to project

This should be reviewed in priority and pulled by everyone

### CODE REVIEW

File: app/src/androidTest/java/com/sdp/swiftwallet/presentation/signIn/RegisterActivityTest.java [hunk 1]
```diff
@@ -27,29 +27,39 @@
...
+    @Inject FirebaseAuth db;
+
+    // Add testing rules
     public ActivityScenarioRule<RegisterActivity> registerScenario = new ActivityScenarioRule<>(RegisterActivity.class);
 
+    public HiltAndroidRule hiltRule = new HiltAndroidRule(this);
+
+    @Rule
+    public final RuleChain rule =
+        RuleChain.outerRule(hiltRule).around(registerScenario);
```
Comment (0xdieman) [inline app/src/androidTest/java/com/sdp/swiftwallet/presentation/signIn/RegisterActivityTest.java:52]: Not sure how that works, I would really appreciate if you could give us a course about hilt ?
Like maybe with a powerpoint.
And like a presentation video if possible.
Also is it possible to do this in english and swedish ?
Jokes aside let's talk about it later

File: app/src/androidTest/java/com/sdp/swiftwallet/uiTest/TransactionActivityTest.java [hunk 2]
```diff
@@ -1,35 +1,11 @@
...
-
-import java.util.ArrayList;
-import java.util.List;
-
+package com.sdp.swiftwallet.uiTest;
 
+/**
+@HiltAndroidTest
 @RunWith(AndroidJUnit4.class)
 public class TransactionActivityTest {
```
Comment (0xdieman) [inline app/src/androidTest/java/com/sdp/swiftwallet/uiTest/TransactionActivityTest.java:6]: We might want to try for next time to have the custom hilt runner launch an implementation of the same class as the SwiftWalletApp. Otherwise the entire purpose of the app "context" is lost.

File: app/src/main/java/com/sdp/swiftwallet/common/FirebaseAuthModule.java [hunk 3]
```diff
@@ -1,17 +1,31 @@
...
 @Module
-@InstallIn(ActivityComponent.class)
+@InstallIn(SingletonComponent.class)
 public class FirebaseAuthModule {
+
+    /**
+     * For hilt
+     */
+    @Inject
+    public FirebaseAuthModule(){ }
```
Comment (0xdieman) [inline app/src/main/java/com/sdp/swiftwallet/common/FirebaseAuthModule.java:20]: Why is this necessary ?

**Review by 0xdieman — APPROVED** [review]
I like hilt

File: app/src/androidTest/java/com/sdp/swiftwallet/uiTest/TransactionActivityTest.java (hunk 2 above)
Comment (ahominal) [reply app/src/androidTest/java/com/sdp/swiftwallet/uiTest/TransactionActivityTest.java:6]: Yep that's the plan ! I think we can put it as a task

File: app/src/main/java/com/sdp/swiftwallet/common/FirebaseAuthModule.java (hunk 3 above)
Comment (ahominal) [reply app/src/main/java/com/sdp/swiftwallet/common/FirebaseAuthModule.java:20]: Oh misspell thanks for noticing !

File: app/src/androidTest/java/com/sdp/swiftwallet/presentation/signIn/RegisterActivityTest.java (hunk 1 above)
Comment (ahominal) [reply app/src/androidTest/java/com/sdp/swiftwallet/presentation/signIn/RegisterActivityTest.java:52]: RuleChain is very complicated, its a composition of multiple rules in a certain order !!

File: app/src/main/java/com/sdp/swiftwallet/common/FirebaseAuthModule.java (hunk 3 above)
Comment (0xdieman) [reply app/src/main/java/com/sdp/swiftwallet/common/FirebaseAuthModule.java:20]: hihi
